test/ngMessageFormat/messageFormatSpec.js

```diff
@@ -0,0 +1,699 @@
+'use strict';
+
+/* TODO: Add tests for:
+ • Whitespace preservation in messages.
+ • Whitespace ignored around syntax except for offset:N.
+ • Escaping for curlies and the # symbol.
+ • # symbol value.
+ • # symbol value when gender is nested inside plural.
```

Comment:
This seems like quite an important use case. I guess the actual code does already work. We should prioritize this test after we get this merged.